src/appMethods.ts

```diff
@@ -223,20 +211,17 @@ export const sponsor = async (
 
   const message = getMessage(op);
   const arrayedMessage = Buffer.from(message);
-
+  const arrayedKey = B64.toByteArray(key);
+  const signature = nacl.sign.detached(arrayedMessage, arrayedKey);
```

Comment:
You tested this right?  (Do we have tests 😄 ?)

Reply:
there are no tests, but I ran the function manually before committing and it worked